src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2EditionSets.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  Box,
+  Flex,
+  RadioGroup,
+  Radio,
+  Separator,
+  Text,
+  FlexProps,
+} from "@artsy/palette"
+import { createFragmentContainer, graphql } from "react-relay"
+
+import { ArtworkSidebar2EditionSets_artwork } from "__generated__/ArtworkSidebar2EditionSets_artwork.graphql"
+import React from "react"
+import { ArtworkSidebar2SizeInfoFragmentContainer } from "./ArtworkSidebar2SizeInfo"
+
+const Row: React.FC<FlexProps> = ({ children, ...others }) => (
+  <Flex justifyContent="left" {...others}>
+    {children}
+  </Flex>
+)
+
+type EditionSet = NonNullable<
+  ArtworkSidebar2EditionSets_artwork["editionSets"]
+>[0]
+
+interface ArtworkSidebar2EditionSetsProps {
+  artwork: ArtworkSidebar2EditionSets_artwork
+  selectedEditionSet?: EditionSet
+  onSelectEditionSet?: (editionSet: EditionSet) => void
+}
+
+const ArtworkSidebar2EditionSets: React.FC<ArtworkSidebar2EditionSetsProps> = ({
+  artwork,
+}) => {
+  const { editionSets, isInquireable, isOfferable, isAcquireable } = artwork
+
+  const renderEditionSet = (editionSet: EditionSet) => {
+    const editionEcommerceAvailable =
+      editionSet?.isAcquireable || editionSet?.isOfferable || isInquireable
+
+    const editionFragment = (
+      <Flex justifyContent="space-between" flex={1}>
+        <ArtworkSidebar2SizeInfoFragmentContainer piece={editionSet!} />
+
+        <Text
+          ml={1}
+          variant="sm-display"
+          color="black100"
+          data-test="SaleMessage"
+        >
+          {editionSet?.saleMessage}
+        </Text>
+      </Flex>
+    )
+
+    if (!!(isOfferable || isAcquireable || isInquireable)) {
+      return (
+        <Row>
+          <Radio
+            flex={1}
+            onSelect={() => {
+              // TODO: will be added in the next pr
+              // this.setState({ selectedEditionSet: editionSet })
+            }}
+            selected={
+              // TODO: will be added in the next pr
+              false
+            }
+            disabled={!editionEcommerceAvailable}
+            label={editionFragment}
+          />
+        </Row>
+      )
+    } else {
+      return <Row>{editionFragment}</Row>
+    }
+  }
+
+  return (
+    <RadioGroup>
+      {
+        editionSets?.map((editionSet, index) => (
+          <React.Fragment key={editionSet?.id}>
+            <Box py={2}>{renderEditionSet(editionSet)}</Box>
+            {index !== editionSets.length - 1 && <Separator />}
```

Comment:
**question**: maybe we can use [Join](https://palette.artsy.net/elements/join/) component here?

Reply:
Unfortunately we can't use join here since the separator from Join has to be an element, and in our case when the condition in line 85 is false we don't render one and we get this type error:

`Type 'Element | undefined' is not assignable to type 'ReactElement<any, string | JSXElementConstructor<any>>'.`
or
`Type 'Element | null' is not assignable to type 'ReactElement<any, string | JSXElementConstructor<any>>'.`

Reply:
got it. thanks for checking!